Fix OpenAPI enricher config parsing

mslabek opened the pull request:
## Describe your changes
Previously there was a bug where `secret` / `secrets` fields weren't parsed at all. Tests were using different parsing method, so this behaviour wasn't visible in tests.

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

mslabek committed:
fix openapi enricher config parsing

philemone reviewed (approved):
Awesome! Can we also add some note in `OpenAPI.md` about the way we read this config? I mean sth like
> When no `security` is defined in the config the first secret matching expected security schema type will be used'

WDYT? does it make sense?

mslabek committed:
add docs entry and test

mslabek commented:
> Awesome! Can we also add some note in `OpenAPI.md` about the way we read this config? I mean sth like
> 
> > When no `security` is defined in the config the first secret matching expected security schema type will be used'
> 
> WDYT? does it make sense?

I've added a `Configuring authentication` doc section that should clarify this. Its hard to describe, so I used an example. I've also added a test that checks if `security` takes precedence over `secret` if both match, since I wanted to add it to the doc

mslabek committed:
update doc
